domains/client/domain-executor/src/sortition.rs

```diff
@@ -0,0 +1,294 @@
+//! Domain sortition related logic.
+
+use codec::Encode;
+use domain_runtime_primitives::DomainCoreApi;
+use sp_api::ProvideRuntimeApi;
+use sp_runtime::traits::Block as BlockT;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use subspace_core_primitives::crypto::blake2b_256_hash;
+use subspace_core_primitives::{bidirectional_distance, U256};
+
+const INITIAL_TX_RANGE: NonZeroUsize = NonZeroUsize::new(10).expect("Not zero; qed");
+
+/// Ttransaction sortition for inclusion in a proposed bundle.
```

Comment:
```suggestion
/// Transaction sortition for inclusion in a proposed bundle.
```

Reply:
Just chipping in on initial tx range: as discussed at the offsite with setting the initial range to 10% of all pool we expect (with high probability) all transactions cleared in ~10 slots (see [notes](https://www.notion.so/subspacelabs/Efficient-Transaction-Sortition-across-bundles-88f8fb2215224316a487815df3d41e86?pvs=4#3b3ee6a92e0149319141e551c476fb49)). However, this is an assumption. We can tweak it if it proves not to be adequate in practice. Notice also that the initial range in practice will only be for the first 10 min and will be dynamically adjusted every following 600 slots.